pallets/did/src/lib.rs

```diff
@@ -749,6 +786,11 @@ pub mod pallet {
 			let did_subject = T::EnsureOrigin::ensure_origin(origin)?.subject();
 			let mut did_details = Did::<T>::get(&did_subject).ok_or(Error::<T>::NotFound)?;
 
+    // Check if the max limit for key agreement keys has been exceeded
```

Comment:
Not required to add this, the error is already present. Please check `did_details` file.